### PR TITLE
[drci] fix cancelled workflows showing up

### PR DESCRIPTION
**Pull request by clee2000**
change sql query to get all workflows (previously got only the ones that didnt have any jobs), then the code that removes reruns removes the workflows that are rerun, then remove workflows if there is a job belonging to the workflow -> no double counting

### CODE REVIEW

File: torchci/rockset/commons/__sql/recent_pr_workflows_query.sql [hunk 1]
```diff
@@ -15,32 +16,30 @@ WITH recent_shas as (
...
+    recent_shas.sha as head_sha,
     j.torchci_classification.captures as failure_captures,
 FROM
     recent_shas
     join commons.workflow_job j ON j.head_sha = recent_shas.sha
-    left outer join commons.pull_request p on p.head.sha = j.head_sha
+    join commons.workflow_run w on w.id = j.run_id
 UNION
 SELECT
+    null as workflow_name,
```
Comment (ZainRizvi) [inline torchci/rockset/commons/__sql/recent_pr_workflows_query.sql:34]: why?

File: torchci/pages/api/drci/drci.ts [hunk 2]
```diff
@@ -82,6 +82,7 @@ export async function updateDrciComments(octokit: Octokit, prNumber?: string) {
 
       await updateCommentWithWorkflow(octokit, pr_info, comment);
     });
+    console.log("done")
```
Comment (huydhn) [inline torchci/pages/api/drci/drci.ts:85]: Redundant debugging log?

**Review by huydhn — APPROVED** [review]
LGTM!

File: torchci/pages/api/drci/drci.ts [hunk 3]
```diff
@@ -270,15 +271,30 @@ export function reorganizeWorkflows(
...
-    const existing_job = workflowsByPR.get(pr_number).jobs.get(name);
+    const existing_job = workflowsByPR.get(pr_number)?.jobs.get(name);
     if (!existing_job || existing_job.id < workflow.id!) {
       // if rerun, choose the job with the larger id as that is more recent
-      workflowsByPR.get(pr_number).jobs.set(name, workflow);
+      workflowsByPR.get(pr_number)!.jobs.set(name, workflow);
     }
   }
+
+  // clean up the workflows - remove workflows that have jobs
```
Comment (ZainRizvi) [inline torchci/pages/api/drci/drci.ts:285]: Why do you want to remove workflows that have jobs?

What's the problem that this is trying to solve?

**Review by ZainRizvi — COMMENTED** [review]
> then remove workflows if there is a job belonging to the workflow -> no double counting

It's not quite clear what the bug is that you're referring to here. Mind elaborating?

File: torchci/rockset/commons/__sql/recent_pr_workflows_query.sql (hunk 1 above)
Comment (clee2000) [reply torchci/rockset/commons/__sql/recent_pr_workflows_query.sql:34]: union requires that the number of fields be the same.  This is the part of the query that finds the workflows and the name of the workflow is stored in the name field

File: torchci/pages/api/drci/drci.ts (hunk 3 above)
Comment (clee2000) [reply torchci/pages/api/drci/drci.ts:285]: if you dont, you get lists like "Lint, lintrunner, toc failed", except Lint is the name of the workflow, which is usually not preferred when you could just get the actual jobs that failed (lintrunner and toc in this case)

**Pull request merged**